src/Adapter/MSTest.CoreAdapter/TestMethodFilter.cs

```diff
@@ -117,7 +130,7 @@ private ITestCaseFilterExpression GetTestCaseFilterFromDiscoveryContext(IDiscove
             {
                 // GetTestCaseFilter is present in DiscoveryContext but not in IDiscoveryContext interface.
                 MethodInfo methodGetTestCaseFilter = context.GetType().GetRuntimeMethod("GetTestCaseFilter", new[] { typeof(IEnumerable<string>), typeof(Func<string, TestProperty>) });
-                return (ITestCaseFilterExpression)methodGetTestCaseFilter?.Invoke(context, new object[] { this.supportedProperties.Keys, (Func<string, TestProperty>)this.PropertyProvider });
+                return SafeInvoke(() => methodGetTestCaseFilter?.Invoke(context, new object[] { this.supportedProperties.Keys, (Func<string, TestProperty>)this.PropertyProvider })) as ITestCaseFilterExpression;
             }
             catch (TargetInvocationException ex)
```

Comment:
Can we make this a catch(Exception) and log and bail out instead?
What is the scenario and the exception being thrown here?

Reply:
The exception is thrown in UWP release with .net native compilation being enabled.
The exception is "Arg_NotImplementedException" which means that the request method was not implemented. https://github.com/dotnet/corert/blob/35f68948d5ef585b67f820efd5936091f67b5418/src/System.Private.CoreLib/src/Resources/Strings.resx#L369

I cannot log everything in exception, since that would set "filterHasError" to true & would fail the discovery.

I don't know the root cause of it yet, but I've dropped a mail to UWP team asking why this could happen

I've described the scenario, but I really do not know how to add test for this scenario